platform/solr/solr-password/src/test/groovy/SolrPasswordUpdateSpec.groovy

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+
+import ddf.platform.solr.security.SolrAuthResource
+import ddf.platform.solr.security.SolrPasswordUpdateImpl
+import ddf.security.encryption.EncryptionService
+import org.codice.ddf.cxf.client.ClientFactoryFactory
+import org.codice.ddf.cxf.client.SecureCxfClientFactory
+import org.codice.ddf.platform.util.uuidgenerator.UuidGenerator
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import javax.ws.rs.core.Response
+
+class SolrPasswordUpdateSpec extends Specification {
+
+    public static final String PLAINTEXT_PASSWORD = "plaintext password"
+    public static final String ENCRYPTED_PASSWORD = "encrypted password"
+    public static final String BOOTSTRAP_PASSWORD = "autogenerated"
+    public static final String SOLR_URL = "http://solr"
+    public static final String SOLR_USERNAME = "ddf"
+    private static final String WRAPPED_PASSWORD = String.format("ENC(%s)", ENCRYPTED_PASSWORD);
+
+    def uuidGenerator = Mock(UuidGenerator) {
+        generateUuid() >> PLAINTEXT_PASSWORD
+    }
+
+    def encryptionService = Mock(EncryptionService) {
+        encrypt(PLAINTEXT_PASSWORD) >> ENCRYPTED_PASSWORD
+        decryptValue(BOOTSTRAP_PASSWORD) >> BOOTSTRAP_PASSWORD
+    }
+
+    def setup() {
+        System.setProperty("solr.http.url", SOLR_URL);
```

Comment:
❗️ You should use the `ResetSystemProperties` extension to make sure system properties are reverted after each test